### ISSUE TITLE
Cannot deploy mup

### ISSUE THREAD

**Issue by eportico** [arunoda/meteor-up-legacy#305]
Hi gyus,
i can't deploy meteor app. I get this error:
ndle.onexit (child_process.js:820:12)
        gyp ERR! System Linux 3.16.0-30-generic
        gyp ERR! command "node" "/usr/bin/node-gyp" "rebuild"
        gyp ERR! cwd /opt/reactivebox/tmp/bundle/programs/server/npm/npm-container/node_modules/usage
        gyp ERR! node -v v0.10.35
        gyp ERR! node-gyp -v v1.0.3
        gyp ERR! not ok
        npm WARN cannot run in wd usage@0.5.0 node-gyp rebuild (wd=/opt/reactivebox/tmp/bundle/programs/server/npm/npm-container/node_modules/usage)
        npm WARN package.json meteor-dev-bundle@0.0.0 No description
        npm WARN package.json meteor-dev-bundle@0.0.0 No repository field.
        npm WARN package.json meteor-dev-bundle@0.0.0 No README data
          % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                         Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 8888: Connection refused
        Latest deployment failed! Reverted back to the previous version.

Don't deploy last version and it reverted. Any idea ?

**Comment by jimmy1993k** [comment]
Same problem:

```
gyp info spawn args   '--depth=.',
gyp info spawn args   '--no-parallel',
gyp info spawn args   '--generator-output',
gyp info spawn args   'build',
gyp info spawn args   '-Goutput_dir=.' ]
gyp info spawn make
gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
gyp info ok 
npm WARN cannot run in wd bcrypt@0.7.7 node-gyp rebuild (wd=/opt/<app_name>/tmp/bundle/programs/server/npm/npm-bcrypt/node_modules/bcrypt)
npm WARN package.json meteor-dev-bundle@0.0.0 No description
npm WARN package.json meteor-dev-bundle@0.0.0 No repository field.
npm WARN package.json meteor-dev-bundle@0.0.0 No README data
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
```

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0curl: (7) Failed to connect to localhost port 80: Connection refused
    Latest deployment failed! Reverted back to the previous version.

It worked fine before (don't mind the <app_name>, I put it there for other reasons)

**Comment by arunoda** [comment]
May be this is due to the recent upgrade.
Try to revert the version to 0.9.1

I will try a fix.
What's your server and where did you get it.

**Comment by jimmy1993k** [comment]
Nevermind, it was an error on my side, I checked the logs and there was an error in the code, only caused by old settings. This was not quite clear because of the error message though. Thanks for the quick reply anyway.

**Comment by benjamin79** [comment]
@jimmy1993k  I´ve the same error with one app. What did you change?

**Comment by joncursi** [comment]
To anyone encountering this type of error with bcrypt, we were able to fix it simply by increasing the `deployCheckWaitTime` inside `mup.json` from 15 to 120 - you may need to adjust that number accordingly. Essentially, the Meteor app wasn't restarting fast enough (within the 15 second timebox) and that was causing issues in the mup deploy process.

@benjamin79

**Comment by joncursi** [comment]
@benjamin79 also, make sure you're using the latest version of mup: `npm update mup -g`

**Comment by eportico** [comment]
This error is fixed width new version 0.9.5. Now is fine

Thanks @arunoda

**Comment by shwaydogg** [comment]
For anyone else coming to this page with the same error, another cause is the node version being out of date.  The current default `0.10.36` needed to be changed to `0.10.40`.  This was also in the mup logs though, don't forget `mup logs -f` to tail them!
